src/core/__tests__/ApolloClient/general.test.ts

```diff
@@ -8054,6 +8054,164 @@ describe("ApolloClient", () => {
       }
     );
   });
+
+  describe("forbidden field alias names", () => {
+    it("`ApolloClient.query` throws when encountering a query that aliases another field to `__typename`", async () => {
+      const client = new ApolloClient({
+        cache: new InMemoryCache(),
+        link: ApolloLink.empty(),
+      });
+      const query = gql`
+        query Test {
+          __typename: hello
+        }
+      `;
+      expect(() => client.query({ query })).toThrow(
+        new InvariantError(
+          '`__typename` is a forbidden field alias name in the selection set for field `__typename` in query "Test".'
+        )
+      );
+    });
+
+    it("`ApolloClient.query` throws when encountering a query that aliases another field to `__ac_*`", async () => {
+      const client = new ApolloClient({
+        cache: new InMemoryCache(),
+        link: ApolloLink.empty(),
+      });
+      const query = gql`
+        query Test {
+          foo {
+            __ac_foo: hello
+          }
+        }
+      `;
+      expect(() => client.query({ query })).toThrow(
+        new InvariantError(
+          '`__ac_foo` is a forbidden field alias name in the selection set for field `foo.__ac_foo` in query "Test".'
```

Comment:
```suggestion
          '`__ac_foo` is a forbidden field alias name in the selection set for field `foo.hello` in query "Test".'
```

Same here, would be great if this were the schema name instead of the alias name.